dependency-resolution/index.md

```diff
@@ -102,3 +102,50 @@ If instead `D11` depends on exactly `1.0.0` and all others depend on `^1.0.0` it
 If instead `D11` depends on exactly `1.0.0` and all other depends on `^1.2.0`, `Dx` will be installed twice: once as `1.6.0` at `node_modules/Dx` and once as `1.0.0` at `node_modules/D11/node_modules/Dx`.
 
 If a lockfile is used by the parent project, the contents of the `node_modules/` directory are, and by extension Node.js' dependency resolution is, entirely reproducible (ignoring installation scripts).
+
+## Python
+### pip
+pip is the package installer for Python and included with modern versions of Python, it is used to find, download, and install packages from [PyPI](https://pypi.org/) and other Python package indexes.
+
+The process of package resolution in pip relies on three key pieces of information: `project name`, `release version` and `dependencies`. The first two pieces of information identify an individual “candidate” for installation, and the third one is used on “on demand” when backtracking algorithm starts to check that particular candidate.
+
+When `pip install` is executed, pip first performs a dependency resolution process by analyzing all the dependencies of the requested packages to determine the most compatible version (according to [version specifiers](https://packaging.python.org/en/latest/glossary/#term-Version-Specifier) and the most recent version). The resolution process might involve backtracking to find the best combination that satisfies all dependency constraints. Only one version of a dependency will get installed.
```

Comment:
> most recent version

This could mean most recent in terms of date or semver or some other format. Could you look up what does `most recent` mean? For example, `1.1.0` could be released at a later time than `2.0.0` so which one is most recent?

Reply:
It is a good question! The term used by pip doc is "Latest version"(https://pip.pypa.io/en/stable/cli/pip_install/#satisfying-requirements), and in Python Packaging guide, it says "When multiple candidate versions match a version specifier, the preferred version SHOULD be the latest version as determined by the consistent ordering defined by the standard [Version scheme](https://packaging.python.org/en/latest/specifications/version-specifiers/#version-scheme)." So in theory `2.0.0` should be later than `1.1.0`, but if 1.1.0 is released at a later time than 2.0.0, maybe 1.1.0 is the latest version(not sure, I am confusing)
I will change the description to "Latest version"